Initial commit adding the ability to disable excludes to the yum module.

lil-cain opened the pull request:
Hey,

We use excludes pretty heavily, and the ability to disable them when updating would be pretty nice.

Obviously, we could get around this by writing a plugin to rewrite the yum config, and then fix it post ansible run, or pushing a seperate updates yum.conf, but both ways seem pretty ugly :-)

Cian.

lil-cain committed:
Initial commit adding the ability to disable excludes to the yum module.

mpdehaan commented:
"+  disableexclues:"

Per ansible conventions, this option should be disable_excludes, should not be mispelled, and should have a "version_added" attribute that indicates what version of Ansible this was added in.

The additional "respecting excludes" comment should not be added as it complicates the previous documentation example, and this option is probably simple enough that it doesn't need an explicit example.

Disable excludes is a bit of a double negative, so I'm also wondering if there is a cleaner way to express what you mean to do here.

lil-cain committed:
Fixes for comments on issue #7751

- fix documentation
- ensure 'exclude' is spelt right everywhere it's used
- change disableexcludes to disable_excludes to fit ansible standards
- added 'version_added' to documentation for disable_excludes

lil-cain commented:
Hi,
Removed the documentation example, added a "version_added" field (This should be 1.7, I assume?), spelt disable_excludes properly.

disable excludes is the term yum uses, so is probably the one familiar to users.

Cian.

mpdehaan commented:
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thanks you very much!

solomongifford commented:
lil-cain - was this ever applied to the ansible-modules-core repo?

lil-cain commented:
No. I haven't had a chance to resubmit it yet. Will try and get it done this evening.

lil-cain commented:
Hi,

Forgot to comment here, but put this pull request into the ansible-modules-core repo:

https://github.com/ansible/ansible-modules-core/pull/307

Cian
